### DISCUSSION TITLE
Can We Bring Our Own Model?

### DISCUSSION THREAD

**Discussion by RudyVenguswamy** [PyTorchLightning/pytorch-lightning-bolts#357] [Q&A] [answered]
**Context:** We’re a team trying to build a reverse image searcher for satellite data. Currently, we’re using the UC Merced land use dataset (linked in notebook below) to train a self-supervised learner and evaluate it using the labels provided with the dataset. 

**Question:** Training a SimCLR with Resnet50bn takes a lot of FP operations. Is it possible to implement or pass our own encoder to SimCLR (that isn't a resnet)? It seems the projection head allows us to pass in the number of dimensions of the output of the encoder. Should we just simply override the model.encoder, or will this lead to issues with model architecture? How might we do so for other SSL models as well?

**Comment by ananyahjha93** [comment, marked as answer]
You can override the encoder by reassigning model.encoder. 

the second option is to subclass SimCLR model and override just the init_encoder() function.
